src/NuGet.Clients/PackageManagement.VisualStudio/IDE/VSSolutionManager.cs

```diff
@@ -1004,7 +1009,40 @@ internal static IEnumerable<EnvDTE.Project> GetDependentEnvDTEProjects(IDictiona
             dependentEnvDTEProjects.Add(dependentEnvDTEProject);
         }
 
-#region IVsSelectionEvents
+        /// <summary>
+        /// This method is invoked when ProjectSystemCache fires a CacheUpdated event.
+        /// This method inturn invokes AfterNuGetCacheUpdated event which is consumed by PackageManagerControl.xaml.cs
+        /// </summary>
+        /// <param name="sender">Event sender object</param>
+        /// <param name="e">Event arguments. This will be EventArgs.Empty</param>
+        private void NuGetCacheUpdate_After(object sender, EventArgs e)
+        {
+            // The AfterNuGetCacheUpdated event is raised on a separate Task to prevent blocking of the caller.
+            // E.g. - If Restore updates the cache entries on CPS nomination, then restore should not be blocked till UI is restored.
+            NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(() => FireNuGetCacheUpdatedEventAsync(e));
```

Comment:
I looked at this with @mishra14 earlier, it can go on to the UI thread, but it is in a different JTF then. Task.Run could be used here I think, but it is unclear if it would make a difference. @jainaashish do you know if there is a perf difference?